lib/faraday/adapter/net_http.rb

```diff
@@ -89,11 +89,15 @@ def with_net_http_connection(env)
       end
 
       def net_http_connection(env)
-        if proxy = env[:request][:proxy]
-          Net::HTTP::Proxy(proxy[:uri].host, proxy[:uri].port, proxy[:user], proxy[:password])
+        proxy = env[:request][:proxy]
+        addr  = env[:url].host
+        port  = env[:url].port || (env[:url].scheme == 'https' ? 443 : 80)
+        # HTTP.Proxy is obsolete, and Net::HTTP will use proxy from env as default
+        if proxy.nil?
+          Net::HTTP.new(addr, port, nil)
```

Comment:
Sending `nil` as the 3rd parameter we change the default behaviour (use the env proxy) and we also make it impossible to enable it back.
This is not acceptable as it could break existing code.